projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts

```diff
@@ -125,8 +187,10 @@ export class DocumentDetailViewComponent implements DetailRecord, OnInit, OnDest
     const keysTypes = Object.keys(this.mappingtypes);
     const rec = record.metadata;
     const route = ['/records', 'documents', 'new'];
+    const queryParams = [];
+    // If we have identifiers
     if (rec.identifiedBy) {
-      const queryParams = [];
+      // We select only a defined part of the identifier
       rec.identifiedBy.forEach((identifier: any) => {
```

Comment:
```typescript
private _my_identifier_list = [IdentifierType.DOI, IdentifierType.ISSN, ...];  // use a more explicit name. I havn't idea... 
...

rec.identifiedBy
    .filter((identifier: any) => this._my_identifier_list.includes(identifier.type))
    .forEach((identifier: any) => {
        queryParams.push(this._extractAndFormatQueryParams(identifier));
        if (identifier.type in this.mappingtypes) { // make `mappingtypes` private and snake case
            const cidentifier = cloneDeep(identifier);
            cidentifier.type = this.mappingtypes[cidentifier.type];
            queryParams.push(this._extractAndFormatQueryParams(cidentifier));
       }
    });
```
Seems more readable for me. I know it's not part of your change.... you can say "no"